lib.php

```diff
@@ -0,0 +1,41 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package auth_saml2
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @copyright Catalyst IT
+ */
+
+/**
+ * Check if we have the saml=on param set. If so, disable guest access and force the user to log in with saml.
+ *
+ * @return boolean
+ */
+function auth_saml2_after_config() {
```

Comment:
this hook is very raw and can be a little dangerous in edge cases just as during install or upgrade, run from cli or webservice etc. Can you wrap this in a generic try catch as this should never fail for any reason, but it can log debugging calls

Reply:
@brendanheywood  I've wrapped it all in a try catch now